lookout/style/typos/corrector.py

```diff
@@ -77,118 +82,88 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
 
-    def train_on_file(self, typos_file: str, candidates_file: str = None,
-                      save_candidates_file: str = None) -> None:
-        """
-        Train corrector on the given dataset of typos inside identifiers.
-
-        :param typos_file: CSV file with columns "typo" and "identifier",
-                           column "token_split" is optional, but used when present.
-        :param candidates_file: Pickle dump of pandas.DataFrame with precalculated \
-                                candidates and features
-        :param save_candidates_file: Path to file where to save the candidates.
-        """
-        typos = pandas.read_csv(typos_file, index_col=0)
-        candidates = None
-        if candidates_file is not None:
-            candidates = pandas.read_pickle(candidates_file)
-        self.train(typos, candidates, save_candidates_file)
-
-    def suggest(self, typos: pandas.DataFrame, candidates: pandas.DataFrame = None,
+    def suggest(self, data: Union[pandas.DataFrame, str],
+                candidates: Union[pandas.DataFrame, str] = None,
                 save_candidates_file: str = None, n_candidates: int = 3,
                 return_all: bool = True) -> Dict[int, List[Tuple[str, float]]]:
         """
         Suggest corrections for given typos.
 
-        :param typos: DataFrame containing column "typo", \
-                      column "token_split" is optional, but used when present
-        :param candidates: DataFrame with precalculated candidates
-        :param n_candidates: Number of most probable candidates to return
-        :param return_all: False to return suggestions only for corrected tokens
-        :param save_candidates_file: Path to file to save candidates to
+        :param data: DataFrame or its .csv dump, containing column Columns.Token, \
+                     column Columns.Split is optional, but used when present.
+        :param candidates: DataFrame or its pickle dump with precalculated candidates.
+        :param n_candidates: Number of most probable candidates to return.
+        :param return_all: False to return suggestions only for corrected tokens.
+        :param save_candidates_file: Path to file to save candidates to (.pkl).
         :return: Dictionary {id : [[candidate, correctness_proba]]}, candidates are sorted \
                  by correctness probability in a descending order.
         """
-        if candidates is None:
-            candidates = self.generator.generate_candidates(typos, self.threads_number,
-                                                            save_candidates_file)
+        if isinstance(data, str):
+            data = pandas.read_csv(data, index_col=0)
+        if not candidates:
+            candidates = self.generator.generate_candidates(
+                data, self.threads_number, save_candidates_file)
+        elif isinstance(candidates, str):
+            candidates = pandas.read_pickle(candidates)
```

Comment:
why do you use not only csv files for everything?

Reply:
Well, for historical reasons:) Also `candidates` df is usually uite massive, several times bigger than `data`. But I'm thinking about moving to .csv everywhere myself.

Reply:
is it possible to implement in this pr? 
you can use `dataframe.to_csv(..., compression="xz")` to decrease the file size.

Reply:
This is actually a nice suggestion @zurk

Then it is possible to use xzcat/xzless to see inside.

Reply:
Cool, thank you, I will do that.